cranelift/codegen/src/isa/s390x/abi.rs

```diff
@@ -375,30 +375,30 @@ impl ABIMachineSpec for S390xMachineDeps {
                 0
             };
             if let Some(reg) = get_intreg_for_arg(next_gpr) {
-                ret.push(ABIArg::reg(
+                args.push(ABIArg::reg(
                     reg.to_real_reg().unwrap(),
                     types::I64,
                     ir::ArgumentExtension::None,
                     ir::ArgumentPurpose::Normal,
                 ));
             } else {
-                ret.push(ABIArg::stack(
+                args.push(ABIArg::stack(
                     next_stack as i64,
                     types::I64,
                     ir::ArgumentExtension::None,
                     ir::ArgumentPurpose::Normal,
                 ));
                 next_stack += 8;
             }
-            Some(ret.len() - 1)
+            Some(args.args().len() - 1)
         } else {
             None
         };
 
         // After all arguments are in their well-defined location,
         // allocate buffers for all StructArg or ImplicitPtrArg arguments.
-        for i in 0..ret.len() {
-            match &mut ret[i] {
+        for i in 0..args.args().len() {
+            match &mut args.args_mut()[i] {
```

Comment:
This was pre-existing, but I don't see any reason this loop can't be a simple iterator:

```suggestion
        for arg in args.args_mut() {
            match arg {
```